### PR TITLE
Fix NVMLError subclass check

### PR DESCRIPTION
**Pull request by therc**
### What does this PR do?

It should catch NVMLError exceptions correctly.
Because the current code only performs an equality check, any subclass of NVMLError will never match, so the code will always return False.

### Motivation

As reported in #612, the check fails with an exception on older hardware.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

File: nvml/datadog_checks/nvml/nvml.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def __exit__(self, exception_type, exception_value, traceback):
         # Do nothing if the exception is not from pynvml or there is no exception
         if traceback is None:
             return False
-        if exception_type is not pynvml.NVMLError:
+        if not issubclass(exception_type, pynvml.NVMLError):
```
Comment (ofek) [inline nvml/datadog_checks/nvml/nvml.py:54]:
```suggestion
        if not isinstance(exception_type, pynvml.NVMLError):
```

Comment (therc) [reply nvml/datadog_checks/nvml/nvml.py:54]: Thanks, I also found another bug...

**Comment by therc** [comment]
That did the trick and it only spews out two errors now, but they're ugly for unrelated reasons:

```
Error running check nvml: [{"message": "'NvmlCall' object has no attribute 'log'",
 "traceback": "Traceback (most recent call last):
  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/nvml/nvml.py\", line 122, in gather_gpu
    consumption = NvmlCheck.N.nvmlDeviceGetTotalEnergyConsumption(handle)
Error running check nvml: [{"message": "'NvmlCall' object has no attribute 'log'",
 "traceback": "Traceback (most recent call last):
  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/nvml/nvml.py\", line 137, in gather_gpu
    tx_bytes = NvmlCheck.N.nvmlDeviceGetPcieThroughput(handle, pynvml.NVML_PCIE_UTIL_TX_BYTES)
```

NvmlCall inherits from object and thus has no log field. Should it inherit from AgentCheck or something else instead?

**Comment by ofek** [comment]
Do https://github.com/DataDog/integrations-extras/blob/b470681af0579da9fad20a1f3a351621e3861f14/nvml/datadog_checks/nvml/nvml.py#L44-L45

```

    def __init__(self, name, log):
        self.name = name
        self.log = log
```

**Comment by therc** [comment]
I also updated all the calling sites. Unless you're saying there's some other special magic.

**Comment by therc** [comment]
> I also updated all the calling sites. Unless you're saying there's some other special magic.

Now it only logs the first of the two errors, but it does so over and over....

**Comment by therc** [comment]
I flipped it back to issubclass() and it runs as intended (log edited for clarity):
```
2020-06-09 01:08:21 | INFO | Running check nvml
2020-06-09 01:08:21 | INFO | Monitoring kubelet tags at /var/lib/kubelet/pod-resources/kubelet.sock
2020-06-09 01:08:21 | WARN | Unable to execute NVML function: total_energy_consumption: Not Supported
2020-06-09 01:08:21 | WARN | Unable to execute NVML function: pci_through: Not Supported
2020-06-09 01:08:21 | INFO | Done running check nvml
2020-06-09 01:08:36 | INFO | Running check nvml
2020-06-09 01:08:36 | INFO | Done running check nvml
2020-06-09 01:08:51 | INFO | Running check nvml
2020-06-09 01:08:51 | INFO | Done running check nvml
```

**Pull request merged**